src/dotnet/CarbonAware.WebApi.Tests/unitTests/SciScoreControllerTests.cs

```diff
@@ -0,0 +1,136 @@
+
+using System.Collections.Generic;
+using System.Net;
+using CarbonAware.Model;
+using CarbonAware.WebApi.Controllers;
+using CarbonAware.WebApi.Models;
+using Microsoft.AspNetCore.Mvc;
+using NUnit.Framework;
+
+namespace CarbonAware.WepApi.UnitTests;
+#pragma warning disable CS8618 // Non-nullable field must contain a non-null value when exiting constructor. Consider declaring as nullable.
+
+/// <summary>
+/// Tests that the Web API controller handles and packages various responses from a plugin properly 
+/// including empty responses and exceptions.
+/// </summary>
+[TestFixture]
+public class SciScoreControllerTests : TestsBase
+{
+
+    /// <summary>
+    /// Tests that successfull call to the aggregator with any data returned results in action with OK status.
+    /// </summary>
+    [TestCase(LocationType.Geoposition)]
+    [TestCase(LocationType.CloudProvider)]
+    public async Task SuccessfulCallReturnsOk_MarginalCarbonIntensity(LocationType locationType)
+    {
+        double data = 0.7;
+        var controller = new SciScoreController(this.MockSciScoreLogger.Object, CreateSciScoreAggregator(data).Object);
+        LocationInput location = new LocationInput()
+            { 
+                LocationType = locationType.ToString()
+            };
+        string timeInterval = "2007-03-01T13:00:00Z/2007-03-01T15:30:00Z";
```

Comment:
You've got a couple constants in here. You may want to pull them out into private variables on a constants file.

Reply:
given that each value is only used once or twice for these tests, probably will leave these in the test file.